lms/models/lti_user.py

```diff
@@ -38,15 +38,25 @@ def h_user(self):
     @property
     def is_instructor(self):
         """Whether this user is an instructor."""
-        return any(
-            role in self.roles.lower()
-            for role in ("administrator", "instructor", "teachingassistant")
+        # We consider admins to be instructors for authorization purposes
+        return self.is_admin or any(
+            # And any instructor in the course
+            role.type == RoleType.INSTRUCTOR and role.scope == RoleScope.COURSE
+            for role in self.lti_roles
         )
 
     @property
     def is_learner(self):
         """Whether this user is a learner."""
-        return "learner" in self.roles.lower()
+        return any(
+            role.type == RoleType.LEARNER and role.scope == RoleScope.COURSE
+            for role in self.lti_roles
+        )
+
+    @property
+    def is_admin(self):
+        """Whether this user is an admin."""
+        return any(role.type == RoleType.ADMIN for role in self.lti_roles)
```

Comment:
So here we're changing these methods from using `self.roles` (the raw LTI roles string) over to `self.lti_roles` (the list of parsed `LTIRole` objects from the DB).

I wonder if `raw_roles` might be a better name for the raw `roles` string but that's outside of this PR